docker-ci.sh

```diff
@@ -14,7 +14,14 @@ function run_lint {
 
 function run_ts {
     npm i
-    npx gulp ts-check npm-dts-check
+    npm run update-ts
+
+    if [git status --porcelain ]; then
```

Comment:
Symmetry 😢

Reply:
I'll delete these lines. All hail the Symmetry!

Reply:
> I tried git status but it always returns 0, even if there are items to commit
https://stackoverflow.com/q/5139290

This should work (not tested):
```bash
if [ -n $(git status ts/*.d.ts --porcelain) ]; then
```

Reply:
I've moved the check away from the docker scripts to the gulp file so that it can be run locally: https://github.com/DevExpress/DevExtreme/pull/8055/commits/64424d05a8b3cbb23566eed56449ba0acb0eb5a1

Reply:
See my https://github.com/DevExpress/DevExtreme/pull/8055#discussion_r284663862

I don't see the point of this gulp task. We agreed that contributors run `update-ts` and check for changes via git. Drone can do the same (see above).

Reply:
> Maybe comparing file's content will be enough

Yes, please try this way. Save a backup of d.ts, then check via [`cmp`](https://stackoverflow.com/a/12900693/65899).

Reply:
I think `diff -U` fits better

Reply:
My idea was to use `cmp --silent`. If output is important then `diff` is fine.

Reply:
Here it is: [failed check](https://devextreme-ci.devexpress.com/DevExpress/DevExtreme/16192/19), [passed check](https://devextreme-ci.devexpress.com/DevExpress/DevExtreme/16193/19).

Reply:
It's OK. However, I'd prefer to rely on the exit code:

```bash
DTS=./ts/dx.all.d.ts

if ! diff $DTS.current $DTS -U 5 > $DTS.diff; then
    echo "FAIL: $DTS is outdated:"
    cat $DTS.diff
fi
```

Reply:
[done](https://github.com/DevExpress/DevExtreme/pull/8055/files#diff-4851192b271a019436fa8db6807c9daeR15)